modules/signals/spec/state-source.spec.ts

```diff
@@ -294,6 +294,27 @@ describe('StateSource', () => {
         TestBed.tick();
         expect(executionCount).toBe(2);
       });
+
+      it('does not support a dynamic type as state', () => {
+        const Store = signalStore(
+          { providedIn: 'root' },
+          withState<Record<number, number>>({}),
+          withMethods((store) => ({
+            addNumber(num: number): void {
+              patchState(store, {
+                [num]: num,
+              });
+            },
+          }))
+        );
+        const store = TestBed.inject(Store);
+
+        store.addNumber(1);
+        store.addNumber(2);
+        store.addNumber(3);
+
+        expect(getState(store)).toEqual({});
+      });
```

Comment:
this test already exists. search for: "does not support a dynamic dictionary as state"
```suggestion
```

Reply:
Thanks